HUDOC.js

```diff
@@ -0,0 +1,502 @@
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-01-23 13:29:01"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 JONAS SKORZAK
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+/*
+
+TODO:
+- Handle friendly settlements (the addition to the case name)
+- Handle more than judgments or decisions (summaries, translations, search results)
+- Handle reports
+- Handle resolutions and advisory opinions properly (esp. title)
+- Handle different types of documents via the API.
+- Use keywords for tags
+- Use references to create "related" entries if they exist
+- Put in short titles
+
+=> All relate to one thing: This translator is currently unable to access the API
+to query it for the taxonomies/thesauruses used for decision type ("typedescription"),
+decision-making body ("originatingbody"), keywords ("kpthesaurus"), ...
+Check compiled.js on the HUDOC website in order to learn how to query these aspects.
+Search for "// ###..//sites/echr/echr.js ###"" in compiled.js to find most API fields.
+*/
+
+function text(docOrElem, selector, index) { var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.textContent: null; }
+
+// Scrapes some metadata from the document
+// TODO: integrate function into scrape
+function scrapeMetaData(doc, detail) { // Only scrapes the header of the main page
+	switch (detail) {
+		case "appno": return text(doc, "span.column01");
+		case "typedescription": return text(doc, "span.column02");
+		case "originatingbody": return text(doc, "span.column03");
+		case "judgementdate": return text(doc, "span.column04");
+		default: return null;
+	}
+}
+
+// Gets the itemid of the document
+function getItemID(url) {
+	var urlRegex = /%22itemid%22:\[%22([\d-]*)%22]|\?i=([\d-]*)/i;
+	var id = url.match(urlRegex);
+	return id[1];
+}
+
+// Adds the type of judgment at the end of the name
+function getTypeBit(doc, url) { // TODO: Switch to 'url' once we use the API instead
+	var description = scrapeMetaData(doc, "typedescription");
+	var language = scrapeMetaData(doc, "typedescription");
+
+	//Simpler logic for French that just takes over the type (unless it's a judgment)
+	// TODO: Add comprehensive logic like for English
```

Comment:
This is a big TODO. We shouldn't include the French site in `target` until it translates equivalently to the English site, because silently returning an incomplete item is confusing for users.

Reply:
I believe it is not adequately covered